tmt/result.py

```diff
@@ -189,6 +189,27 @@ class CheckResult(BaseResult):
         default=CheckEvent.BEFORE_TEST,
         serialize=lambda event: event.value,
         unserialize=CheckEvent.from_spec)
+    check_result: CheckResultInterpret = field(
```

Comment:
The result class cannot contain a field that governs its own interpretation. That is owned by a different entity: `Test.result` defines how `Result` should be interpreted, similarly `Check.result` defines how `CheckResult` should be interpreted. This field should not exist, check result simply cannot decide itself, it **must** be **told** how to interpret the value.

```python
    def interpret_check_result(self, interpret: CheckResultInterpret) -> 'CheckResult':
        """Interpret check result according to the check_result interpretation."""

        if interpret == CheckResultInterpret.RESPECT:
            return self

        if interpret == CheckResultInterpret.INFO:
            self.result = ResultOutcome.INFO
        elif interpret == CheckResultInterpret.XFAIL:
            self.result = {
                ResultOutcome.FAIL: ResultOutcome.PASS,
                ResultOutcome.PASS: ResultOutcome.FAIL
                }.get(self.result, self.result)

        return self
```

Take a `CheckResult` instance, find the corresponding `Check`, most likely from the `invocation.test.check` field, and pass its `result` to `CheckResult.interpret_check_result()`. Recall the mapping I proposed last week to be put into `Result.interpret_result()`:

```python
{
    check.how: check.result
    for check in invocation.test.check
}
```

It's the same solution, the idea is there: the interpretation **does not** lie in the result itself, it's owned by test (for test results) or test check (for test check results), and must be extracted from them (which is exactly what `Result.from_test_invocation()` does, `return _result.interpret_result(invocation.test.result)`).